app/src/main/java/com/example/evently/data/EventsDB.java

```diff
@@ -332,4 +343,38 @@ private Promise<List<Event>> parseQuerySnapShots(Task<QuerySnapshot> task) {
                 .flatMap(Optional::stream)
                 .collect(Collectors.toList()));
     }
+
+    /**
+     * Uploads a selected poster to firebase
+     * @param eventID the eventID of the poster.
+     * @param uri the uri of the image
+     * @return a promise of the upload task
+     */
+    public Promise<UploadTask.TaskSnapshot> storePoster(UUID eventID, Uri uri) {
+        StorageReference imageRef = storageRef.child("posters/" + eventID.toString());
+
+        var posterStorageTask = imageRef.putFile(uri);
+        return promise(posterStorageTask);
+    }
+
+    /**
+     * The code below loads a poster into an image view, if the poster is found
+     * @param eventID the eventID of the poster
+     * @param context the context of the image view
+     * @param imageView the imageView
+     */
+    public void showPoster(UUID eventID, Context context, ImageView imageView) {
```

Comment:
The `*DB` classes should not touch UI. It could just return the `StorageReference` I suppose.